Source/WebCore/platform/graphics/displaylists/DisplayListRecorder.cpp

```diff
@@ -473,6 +473,8 @@ void Recorder::strokePath(const Path& path)
     auto& state = currentState().state;
     if (state.changes() && state.containsOnlyInlineChanges() && !state.changes().contains(GraphicsContextState::Change::FillBrush) && path.hasInlineData() && path.hasInlineData<LineData>()) {
```

Comment:
Why does this check `FillBrush` and not `StrokeBrush`?

Reply:
It's subtracting potential `FillBrush` changes, so that we only take the optimised path if the current set of changes are (at most) a stroke width, and stroke brush (set to a color, not a pattern/gradient).

Reply:
Makes sense.